### ISSUE TITLE
[feature] 'structure'-files with extra tags

### ISSUE THREAD

**Issue by blekerfeld** [Chistaen/HawkMDP#3]
As php supports anonymous functions, I think they can be stored inside array's as well (not tested), that means that in order to add extra tags to the parser, you could read a file that contains an array with such information, a bit like this

'''
return [
	
	'lemma_link' => [
		'tag' => '//'
		'callback' => function($content){
			return '<a href="?entry/lemma/'.lookUpLemmaID($content).'" class="lemma_link">' . $content . '</a>';
		}
	],

];
'''

Of course the structure of that file depends on how ParseDown works and how you plan using that functionality in Hawk, but this would be a simple way to define extra tags without having to extend on the Hawk class.

**Comment by Chistaen** [comment]
(Yep, apparently you can :P)

Actually, it's possible to go one step further. The original plan for Hawk was to clean up Parsedown's code and to introduce wiki tags, but it might be a better idea to do what you're suggesting for all tags. Hawk would have a library of standard tags stored in a certain file and you'd be able to load tags from a different file as well.

Modularization would also mean that non-standard library tags would be treated the same way as tags in the standard library.

Tl;dr: thanks for the feature request, it's been accepted.
